disnake/onboarding.py

```diff
@@ -97,29 +105,25 @@ class OnboardingPrompt(Hashable):
         If ``False``, the prompt will only appear in community customization.
     """
 
-    __slots__ = (
-        "guild",
-        "id",
-        "title",
-        "options",
-        "single_select",
-        "required",
-        "in_onboarding",
-        "type",
-    )
-
-    def __init__(self, *, guild: Guild, data: OnboardingPromptPayload) -> None:
-        self.guild = guild
-
-        self.id: int = int(data["id"])
-        self.title: str = data["title"]
-        self.options: List[OnboardingPromptOption] = [
-            OnboardingPromptOption(data=option, guild=guild) for option in data["options"]
-        ]
-        self.single_select: bool = data["single_select"]
-        self.required: bool = data["required"]
-        self.in_onboarding: bool = data["in_onboarding"]
-        self.type: OnboardingPromptType = try_enum(OnboardingPromptType, data["type"])
+    __slots__ = ("id", "title", "options", "single_select", "required", "in_onboarding", "type")
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        options: List[OnboardingPromptOption],
+        type: OnboardingPromptType,
+        single_select: bool = False,
+        required: bool = False,
+        in_onboarding: bool = True,
+    ) -> None:
+        self.id: int = int.from_bytes(os.urandom(4))
```

Comment:
Oops, I forgot to remove that 😅

Reply:
> On that note, what do you think about moving this into `to_dict` and keeping `self.id = 0` here? It would match the docstring more closely

Good idea, implemented here c3a7e487b8edb79ba33148c3dab82dde6d6c9b6c